src/servers/Http1Server.cc

```diff
@@ -351,6 +374,21 @@ Http::One::Server::writeControlMsgAndCall(HttpReply *rep, AsyncCall::Pointer &ca
     return true;
 }
 
+void
+switchToTunnel(HttpRequest *request, Comm::ConnectionPointer &clientConn, Comm::ConnectionPointer &srvConn, const SBuf *serverPayload);
```

Comment:
It looks a little strange that we declare this method in two cpp files (this one and ssl/PeekingPeerConnector.cc). Is there a better place for it in a .h file?

Reply:
It would go to a tunnel.h include file but such file does not exist.
I do not know where this function can be declared.

I think I argued in the past about replacing this method with a ConnStateData::splice method at least for the PeekingPeerConnector case. But not sure now about and how this is should be done.

Reply:
What I am trying to say is that the switchToTunnel is probably just a temporary hack.
If it worths to add a new tunnel.h file just for this function we can do it.

Else we can find an other include file, but I do not know which.